airbyte-integrations/connectors/destination-langchain/destination_langchain/processor.py

```diff
@@ -0,0 +1,95 @@
+from typing import List, Optional, Tuple, Mapping
+from destination_langchain.config import ProcessingConfigModel
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+from airbyte_cdk.models import AirbyteRecordMessage, ConfiguredAirbyteCatalog, ConfiguredAirbyteStream
+from langchain.document_loaders.base import Document
+from langchain.utils import stringify_dict
+import dpath.util
+from dpath.exceptions import PathNotFound
+from airbyte_cdk.models.airbyte_protocol import DestinationSyncMode
+import uuid
+from dataclasses import dataclass
+
+METADATA_STREAM_FIELD = "_airbyte_stream"
+METADATA_NATURAL_ID_FIELD = "_natural_id"
+
+class Processor:
```

Comment:
naive question. Why do we need this if embedding APIs can tokenize themselves?

Reply:
Tokenization is just turning utf8 encoded text into a sequence of one-hot vector representations and as you are saying the embedding API takes care of that: https://help.openai.com/en/articles/4936856-what-are-tokens-and-how-to-count-them

What the processor is doing is turning the records into "documents":
* Split them up appropriately
* Separate text and metadata
* Add bookkeeping metadata for incremental sync (id and stream name)